src/pycell.rs

```diff
@@ -366,6 +366,32 @@ where
     /// ```
     pub fn into_super(self) -> PyRef<'p, U> {
         let py = self.py();
+        if <T::Frozen as crate::pyclass::boolean_struct::private::Boolean>::VALUE
+            != <U::Frozen as crate::pyclass::boolean_struct::private::Boolean>::VALUE
```

Comment:
I _think_ we only need to check this when going from mutable to frozen (and not when going from frozen to mutable because that would already use `U`s borrow checker), so maybe `!T::Value && U::Value` would be a tiny optimization.

Reply:
I think that's correct too, have adjusted accordingly 👍